akka-remote/src/main/scala/akka/remote/artery/tcp/SSLEngineProvider.scala

```diff
@@ -164,7 +164,7 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
 
     val engine = sslContext.createSSLEngine(hostname, port)
```

Comment:
Actually, this line is also "wrong" as hostname and port to pass here are `peerHostname` and `peerPort` which makes only sense for the client side.

Reply:
You are correct. However based on the order of execution, it isn't (easi;y) possible to get the client name.